File: sql-traceable/hw3/examples/python/mysql_test_1.py

import mysql.connector as mc
import config

def main():
    try: 
        # connection info
        usr = config.mysql['user']
        pwd = config.mysql['pass']
        hst = config.mysql['host']
        dab = 'bowersDB'
        # create a connection
        con = mc.connect(user=usr,password=pwd, host=hst, database=dab)
        # create a result set
        rs = con.cursor()
        query = 'SELECT id, name FROM pet ORDER BY name'
        # execute the query
        rs.execute(query)
        # print the results
        for (pid, name) in rs:
            print('{}, {}'.format(pid, name))
        
    except mc.Error as err:
        print(err)

    finally:
        rs.close()
        con.close()

if __name__ == '__main__':
    main()
    

File: sql-traceable/hw3/examples/java/MySQLTest1.java

import java.io.FileInputStream;
import java.io.IOException;
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.sql.Statement;
import java.util.Properties;

public class MySQLTest1 {

  public static void main(String[] args) {

    try {
      // connection info
      Properties prop = new Properties();
      FileInputStream in = new FileInputStream("config.properties");
      prop.load(in);
      in.close();
      
      // connect to datbase
      String hst = prop.getProperty("host");
      String usr = prop.getProperty("user");
      String pwd = prop.getProperty("password");
      String dab = "bowersDB";
      String url = "jdbc:mysql://" + hst + "/" + dab;
      Connection cn = DriverManager.getConnection(url, usr, pwd);
      
      // create and execute query
      Statement st = cn.createStatement();
      String q = "SELECT id, name FROM pet ORDER BY name";
      ResultSet rs = st.executeQuery(q);
      
      // print results
      while(rs.next()) {
        String id = rs.getString("id");
        String name = rs.getString("name");
        System.out.println(id + ", " + name);
      }
      
      // release resources
      rs.close();
      st.close();
      cn.close();
    }
    catch(Exception err) {
      err.printStackTrace();
    }
  }
  
}
